service-base/src/main/java/org/eclipse/hono/service/registration/RegistrationService.java

```diff
@@ -47,6 +48,34 @@
      */
     void assertRegistration(String tenantId, String deviceId, Handler<AsyncResult<RegistrationResult>> resultHandler);
 
+    /**
+     * Asserts that a device is registered with a given tenant and is enabled.
+     * <p>
+     * This default implementation simply returns the result of {@link #assertRegistration(String, String, Handler)}.
+     *
+     * @param tenantId The tenant the device belongs to.
+     * @param deviceId The ID of the device to get the assertion for.
+     * @param spanContext The currently active OpenTracing span (may be {@code null}). An implementation
+     *                    should use this as the parent for any span it creates for tracing
+     *                    the execution of this operation.
+     * @param resultHandler The handler to invoke with the result of the operation.
+     *             The <em>status</em> will be
+     *             <ul>
+     *             <li><em>200 OK</em> if a device with the given ID is registered for
+     *             the tenant and its <em>enabled</em> property is {@code true}.
+     *             The <em>payload</em> will contain a JWT token asserting the registration status.</li>
+     *             <li><em>404 Not Found</em> if no device with the given identifier is
+     *             registered for the tenant or its <em>enabled</em> property is {@code false}.</li>
+     *             </ul>
+     * @throws NullPointerException if any of the parameters (except spanContext) is {@code null}.
+     * @see <a href="https://www.eclipse.org/hono/api/device-registration-api/#assert-device-registration">
+     *      Device Registration API - Assert Device Registration</a>
+     */
+    default void assertRegistration(String tenantId, String deviceId, SpanContext spanContext,
```

Comment:
WDYT about deprecating the other variant?

Reply:
Yes, I think it makes sense to only keep the 2 variants with the `SpanContext` parameter in the long run.

Reply:
I think so too but I guess we should wait a while with deprecating them anyway.
Can you please mark the arguments to the default implementations as final?

Reply:
Ok, I've marked them as final.